### ISSUE TITLE
[BUG] RKE2 etcd snapshot from S3 with private CA can not be restored

### ISSUE THREAD

**Issue by spimmer** [rancher/rancher#40994]
**Rancher Server Setup**
- Rancher version: 2.7.1
- Installation option (Docker install/Helm Chart): Helm Chart
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): RKE2
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: v1.24.10+rke2r1 
- Cluster Type (Local/Downstream): Downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): Custom

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom): Admin
  - If custom, define the set of permissions:

**Describe the bug**

When trying to restore an etcd snapshot of an Custom RKE2 cluster from an S3 with a private CA, the restore fails because the CA created on the etcd node is not a valid certificate.

**To Reproduce**

- create a Custom RKE2 cluster with Kubernetes version v1.24 or v1.23
- configure etcd snapshots to be saved to a S3 compatible storage
  - enter the private CA certificate in the cluster config in field "Endpoint CA cert" 
- perform a successful on-demand snapshot
- try to restore the snapshot
  - it does not matter if option "Only etcd", "Kubernetes version and etcd" or "Cluster config, Kubernetes version and etcd" is used
 

**Result**
- in Rancher UI the cluster wil be appear in state "Updating" with message "operation etcd restore failed"
- in log of rancher-system-agent.service you will find the following messages:
  - `Mar 28 03:28:43 down-1-m-1 rancher-system-agent[10039]: time="2023-03-28T03:28:43Z" level=info msg="[c678348e11556aa5028867c9eb88193276f9f83bce29a5b71a151abd5a2871ea_3:stderr]: time=\"2023-03-28T03:28:43Z\" level=fatal msg=\"starting kubernetes: preparing server: start managed database: endpoint-ca is not a valid x509 certificate\""`
  - `Mar 28 03:28:43 down-1-m-1 rancher-system-agent[10039]: time="2023-03-28T03:28:43Z" level=info msg="[Applyinator] Command rke2 [server --cluster-reset --cluster-reset-restore-path=on-demand-down-1-m-1-1679921920 --etcd-s3-bucket=bkp-downstream --etcd-s3-access-key=rancher-bkp --etcd-s3-region=us-east-1 --etcd-s3-endpoint=minio.thanos-poc.sva.wtf:9000 --etcd-s3-endpoint-ca=/var/lib/rancher/rke2/etc/config-files/s3-endpoint-ca-239e6.crt --etcd-s3] finished with err: <nil> and exit code: 1"`
  - `Mar 28 03:28:43 down-1-m-1 rancher-system-agent[10039]: time="2023-03-28T03:28:43Z" level=error msg="error executing instruction 3: <nil>"`
- on a etcd node in folder `/var/lib/rancher/rke2/etc/config-files/` there will be a .crt-file containing a file path and not a valid certificate

**Expected Result**

- the .crt-file used for the restore contains the valid CA certificate
- the cluster will be restored succesfully

**Screenshots**

Content of the crt-files in folder `/var/lib/rancher/rke2/etc/config-files/`:
![grafik](https://user-images.githubusercontent.com/68185774/228177126-8a7f7eb5-e388-4095-9087-b41afcf2b17d.png)

**Additional context**

SURE-6265

**Comment by markusewalker** [comment]
**QA TEST PLAN**

**Scenarios**
|Scenario|Test Case|
|---|---|
|1|As a standard user, provision a downstream node driver RKE2 cluster w/highest K8s version with S3 backup enabled with a private CA, take an etcd snapshot, restore the snapshot and validate that it is successful|
|2|As a standard user, provision a downstream node driver K3S cluster w/highest K8s version with S3 backup enabled with a private CA, take an etcd snapshot, restore the snapshot and validate that it is successful|
|3|As a standard user, provision a downstream node driver K3S cluster w/highest K8s version with S3 backup enabled with any certificate enabled, take an etcd snapshot, restore the snapshot and validate that it is successful|
|4|As a standard user, provision a downstream node driver RKE2 cluster w/second highest K8s version with S3 backup enabled with a private CA, take an etcd snapshot, upgrade cluster's K8s version, restore the snapshot and validate that it is successful|
|5|As a standard user, provision a downstream node driver K3S cluster w/second highest K8s version with S3 backup enabled with a private CA, take an etcd snapshot, upgrade cluster's K8s version, restore the snapshot and validate that it is successful|
|6|Setup Rancher v2.7.3, as a standard user, provision a downstream node driver RKE2 cluster w/highest K8s version with S3 backup enabled with a private CA, upgrade Rancher to v2.7-head, take an etcd snapshot, upgrade cluster's K8s version, restore the snapshot and validate that it is successful|
|7|Setup Rancher v2.7.3, as a standard user, provision a downstream node driver K3S cluster w/highest K8s version with S3 backup enabled with a private CA, upgrade Rancher to v2.7-head, take an etcd snapshot, upgrade cluster's K8s version, restore the snapshot and validate that it is successful|

**Comment by markusewalker** [comment]
Reproduced this on `v2.7.3`. See details below:

**ENVIRONMENT DETAILS**
- Rancher install: Docker
- Rancher version: `v2.7.3`
- Browser: Chrome

**TEST RESULT**
|Scenario|Test Case|Result
|---|---|---|
|1|As a standard user, provision a downstream node driver RKE2 cluster w/highest K8s version with S3 backup enabled with a private CA, take an etcd snapshot, restore the snapshot and validate that it is successful|:x:|

**REPRO STEPS**

**Scenario 1**
1. On a client machine, ran the following commands to stand up MinIO to generate certificates to use later:
```
docker run -v /certs:/certs -e SSL_SUBJECT=<FQDN> -e SSL_DNS=<FQDN> -e SSL_IP=<IP> superseb/omgwtfssl
docker run -d --name minio --restart=unless-stopped -p 9000:9000 -p 9001:9001 -e MINIO_ROOT_USER=<username>-e MINIO_ROOT_PASSWORD=<password>-v /data:/data -v /certs/cert.pem:/etc/minio/tls/public.crt -v /certs/key.pem:/etc/minio/tls/private.key quay.io/minio/minio server /data --certs-dir /etc/minio/tls --console-address ":9001"
```
2. Navigated to MinIO instance and created a bucket and access keysto use when performing backup/restore.
3. As a standard user, provisioned a downstream RKE2 `v1.26.4+rke2r1` node driver cluster.
     - Utilized the created MinIO details, e.g. access keys, bucket, base64 encoded certificate defined earlier.
4. Took an etcd snapshot of the cluster.
5. Attempted to restore the snapshot and reproduced the behavior as shown in the below screen
![image](https://github.com/rancher/rancher/assets/52591819/2d6b635e-2532-4e63-b701-070f46c3a2e5)

**Comment by markusewalker** [comment]
Validated that this is addressed on `v2.7-head` https://github.com/rancher/rancher/commit/c842598344ee552daf5cf4e8d31d6e11621e07ad. See details below:

**ENVIRONMENT DETAILS**
- Rancher install: Docker
- Rancher version: `v2.7-head` https://github.com/rancher/rancher/commit/c842598344ee552daf5cf4e8d31d6e11621e07ad
- Browser: Chrome

**TEST RESULT**
|Scenario|Test Case|Result
|---|---|---|
|1|As a standard user, provision a downstream node driver RKE2 cluster w/highest K8s version with S3 backup enabled with a private CA, take an etcd snapshot, restore the snapshot and validate that it is successful|:white_check_mark:|
|2|As a standard user, provision a downstream node driver K3S cluster w/highest K8s version with S3 backup enabled with a private CA, take an etcd snapshot, restore the snapshot and validate that it is successful|:white_check_mark:|
|3|As a standard user, provision a downstream node driver K3S cluster w/highest K8s version with S3 backup enabled with any certificate enabled, take an etcd snapshot, restore the snapshot and validate that it is successful|:white_check_mark:|
|4|As a standard user, provision a downstream node driver RKE2 cluster w/second highest K8s version with S3 backup enabled with a private CA, take an etcd snapshot, upgrade cluster's K8s version, restore the snapshot and validate that it is successful|:white_check_mark:|
|5|As a standard user, provision a downstream node driver K3S cluster w/second highest K8s version with S3 backup enabled with a private CA, take an etcd snapshot, upgrade cluster's K8s version, restore the snapshot and validate that it is successful|:white_check_mark:|

**VALIDATION STEPS**

**Scenario 1**
1. On a client machine, ran the following commands to stand up MinIO to generate certificates to use later:
```
docker run -v /certs:/certs -e SSL_SUBJECT=<FQDN> -e SSL_DNS=<FQDN> -e SSL_IP=<IP> superseb/omgwtfssl
docker run -d --name minio --restart=unless-stopped -p 9000:9000 -p 9001:9001 -e MINIO_ROOT_USER=<username>-e MINIO_ROOT_PASSWORD=<password>-v /data:/data -v /certs/cert.pem:/etc/minio/tls/public.crt -v /certs/key.pem:/etc/minio/tls/private.key quay.io/minio/minio server /data --certs-dir /etc/minio/tls --console-address ":9001"
```
2. Navigated to MinIO instance and created a bucket and access keysto use when performing backup/restore.
3. As a standard user, provisioned a downstream RKE2 `v1.26.4+rke2r1` node driver cluster.
     - Utilized the created MinIO details, e.g. access keys, bucket, base64 encoded certificate defined earlier.
4. Took an etcd snapshot of the cluster.
5. Successfully restored the etcd snapshot of the cluster using option cluster config, Kubernetes version and etcd.
________
**Scenario 2**
1. Repeated scenario 1, but provisioned a downstream K3S `v1.26.4+k31s` node driver cluster instead.
     - Scenario 2 yielded same successful results as scenario 1.
_____
**Scenario 3**
1. Repeated scenario 1, but provisioned a downstream K3S `v1.26.4+k31s` node driver cluster instead.
     - Additionally, no private CA was used, but rather I enabled accept any certificate in the `Etcd` section while provisioning the cluster.
     - Scenario 3 yielded same successful results as scenario 1.
______
**Scenario 4**
1. Repeated scenario 1 steps 1-3, but provisioned a downstream RKE2 `v1.25.9+rke2r1` instead.
2. Upgraded the cluster to `v1.26.4+rke2r1`; validated that the cluster and nodes are `Active` with upgraded K8s version.
3. Successfully restored the etcd snapshot of the cluster using option cluster config, Kubernetes version and etcd.
     - Verified that the cluster/nodes version is back at `v1.25.9+rke2r1`.
_______
**Scenario 5**
1. Repeated scenario 1 steps 1-3, but provisioned a downstream K3S `v1.25.9+k3s1` instead.
2. Upgraded the cluster to `v1.25.9+k3s1`; validated that the cluster and nodes are `Active` with upgraded K8s version.
3. Successfully restored the etcd snapshot of the cluster using option cluster config, Kubernetes version and etcd.
4. Verified that the cluster/nodes version is back at `v1.25.9+k3s1`.
